packages/cli/src/commands/deploy/index.ts

```diff
@@ -521,9 +521,15 @@ export default async (client: Client): Promise<number> => {
   );
   let nodeVersion: string | undefined;
   if (packageJson?.engines?.node) {
-    const parsedNodeVersion = semver.coerce(packageJson.engines.node);
-    if (parsedNodeVersion) {
-      nodeVersion = `${parsedNodeVersion.major}.x`;
+    try {
+      const { range } = await getSupportedNodeVersion(
+        packageJson?.engines?.node
```

Comment:
Nullish coalescing isn't necessary here since we're already ascertaining in the `if` above:

```suggestion
        packageJson.engines.node
```